docs/DeveloperGuide.md

```diff
@@ -615,66 +637,127 @@ testers are expected to do more *exploratory* testing.
+       Expected: No change in displayed list and employee data. Error details shown in the status message.
+        1. Tip: A quick way to achieve the prerequisite is to close and reopen the application.
 
 ### Redoing the previous undone command
 
 1. Redo the previous undone command caused by `undo`.
 
-   1. Test case: `delete 1` followed by `undo` followed by `redo`<br>
-      Prerequisite: List all persons using the `list` command. At least one person in the list.<br>
-      Expected: The list is reverted to the list of employees after `delete 1` was executed.
-   
-   1. Test case: `redo`<br>
-      Prerequisite: No commands have been undone.<br>
-      Expected: No change in displayed list and employee data. Error details shown in the status message.
+    1. Test case: `delete 1` followed by `undo` followed by `redo`<br>
+       Prerequisite: List all persons using the `list` command. At least one person in the list.<br>
+       Expected: The list is reverted to the list of employees after `delete 1` was executed.
+
+    1. Test case: `redo`<br>
+       Prerequisite: No commands have been undone.<br>
+       Expected: No change in displayed list and employee data. Error details shown in the status message.
+
+### Editing an employee
+
+1. Editing an employee in the displayed list of employees.
+
+    1. Prerequisites: List all employees using the `list` command. At least one employee in the list.
+
+    1. Test case: `edit 1 n/John Tan`<br>
+       Expected: First employee in the list has their name edited to `John Tan`. Details of the edited employee is shown in the status message.
+
+    1. Other correct edit commands to try: `edit 1 p/12345678`, `edit 1 d/Accounting` and `edit 1 r/Manager`<br>
+       Expected: Similar to previous, the respective fields gets edited.
+
+    1. Test case: `edit 0`<br>
+       Expected: No employee is edited. Error details shown in the status message.
+
+    1. Other incorrect edit commands to try: `edit`, `edit x`, `...` (where x is larger than the list size)<br>
+       Expected: Similar to previous.       
+
+### Sorting the list
+
+1. Sorting the displayed list of employees.
+
+    1. Prerequisites: List all employees using the `list` command. Multiple employees in the list.
+
+    1. Test case: `sort n/`<br>
+       Expected: Displayed list is sorted by alphabetical order of the name field.
+
+    1. Other correct sort commands to try: `sort d/`, `sort r/` and `sort e/`<br>
+       Expected: Similar to previous, the respective fields gets sorted based on lexicographical order.
+
+    1. Test case: `sort`<br>
+       Expected: Displayed list is not sorted. Error details shown in the status message.
+
+### Finding an employee
+
+1. Finds an employee by the given field.
+
+    1. Tip: Create distinct employees, one with the name `John`, one with the department `Finance`, 
+   and one with the role `Manager` to effectively test the whether the find command works for each field.
```

Comment:
Perhaps it should be "test whether the find command" rather than "test the whether the find command".